Sender/Generate_data.c

```diff
@@ -0,0 +1,196 @@
+//i.e. ifTrueReadSucess == false signifies the read process failed.
+
+void set_success_stub(bool ifTrueReadSucess)
+{
+  success_stub = ifTrueReadSucess;
+}
+
+bool GenRandomParameter(int *randomNo)
+{
+  *randomNo = 50;
+  return success_stub;
+}
+#endif
+
+
+/* Function to introduce wait cycles or delay in seconds
+* delay(1) --> wait of 1 sec before execution of next command*/
+
+void delay(int number_of_seconds)
+{
+  // Converting time into milli_seconds
+  int milli_seconds = 1000 * number_of_seconds;
+
+  // Storing start time
+  clock_t start_time = clock();
+
+  // looping till required time is not achieved
+  // which is begin time + wait time in milliseconds 
+  while (clock() < start_time + milli_seconds);
+}
+
+read_status GenDataSet(int *temp, int *soc)
+{
+
+  read_status current_status = NotErroneous;
+  //In real scenario we would read from sensor intsead of calling function random number generator 
+  //and we could have communication errors to realize the same, function return indicates if the 
+  // read was successfull 
+
+  bool iSReadSuccessfull_temp = GenRandomParameter(temp);
+  bool iSReadSuccessfull_soc = GenRandomParameter(soc);
+
+  //Signifies communication error while reading from sensor and in such scenarios we send 
+  //NotANum to signify there was error while reading from sensor 
+  if ((!iSReadSuccessfull_temp) && (!iSReadSuccessfull_temp))
+  {
+    current_status = ErroneousPars;
+  }
+  //Signifies communication error while reading from sensor 
+  else if ((!iSReadSuccessfull_soc))
+  {
+    current_status = ErroneousSoc;
+  }
+  else if (!iSReadSuccessfull_temp)
+  {
+    current_status = ErroneousTemp;
+  }
+  return current_status;
+}
+
+void writeDataToFile(int * parameterset)
+{
+  FILE *fptr;
+
+  /* Open file and get the pointer to same, file is either present or created.
+  * The program deletes the file at the end so that everytime sender is run
+  * we write to an empty newfile otherwise it will append at the end of the last dataset.
+  * In any case if the program aborts or the file is not deleted we can do it ourself
+  * and the file will be generated as we begin the program. */
+
+  fptr = fopen("data.txt", "a");
+  if (fptr != NULL)
+  {
+    /* Write both the parameters to file and this will be used to read
+    *  and print on console:
+    *  Format of the generated file temp;soc;\n
+    *  e.g. 50;50;
+    *       50;50;
+    *       ..so on at one time 15 parameter set is generated*/
+
+    /*Print one dataset*/
+
+    for (int Par_Counter = 0; Par_Counter < NoOfPar; Par_Counter++)
+    { 
+      //Print to file Temperature/StateOfCharge parameter
+       fprintf(fptr, "%d", parameterset[Par_Counter]);
+      // Print delimiter which is semicolon
+       fprintf(fptr, "%c", 59);
+    }
+    
+    //Newline as we print one one set in each line
+    fprintf(fptr, "\n");
+
+    //Close file
+    fclose(fptr);
+  }
+  else
+  {
+    // Remark : This statement cannot be tested

```

Comment:
Failure to write can be tested by using a file-path that contains a non-existent folder.